nullaway/src/main/java/com/uber/nullaway/handlers/contract/ContractHandler.java

```diff
@@ -272,24 +271,7 @@ public NullnessHint onDataflowVisitMethodInvocation(
           arg = node.getArgument(i);
           argAntecedentNullness = valueConstraint.equals("null") ? Nullness.NULL : Nullness.NONNULL;
         } else {
-          String errorMessage =
-              "Invalid @Contract annotation detected for method "
-                  + callee
-                  + ". It contains the following uparseable clause: "
-                  + clause
-                  + " (unknown value constraint: "
-                  + valueConstraint
-                  + ", see https://www.jetbrains.com/help/idea/contract-annotations.html).";
-          state.reportMatch(
-              analysis
-                  .getErrorBuilder()
-                  .createErrorDescription(
-                      new ErrorMessage(
-                          ErrorMessage.MessageTypes.ANNOTATION_VALUE_INVALID, errorMessage),
-                      tree,
-                      analysis.buildDescription(tree),
-                      state,
-                      null));
+          // invalid, but we report an error only on method declarations
```

Comment:
Note that we have moved the logic for syntactically validating contracts only when we are also checking that the annotated method satisfies the contract. I think this is fine, but it might lead to lack of user visible information for the cases where they are using contracts in "trust the annotation" mode yet see errors pop up that the contract should suppress, without realizing it is because the contract string is malformed.

I think in modern NullAway it makes sense that if you are adding custom contracts to your codebase you are checking them too and not just trusting, but it is a change we probably want to note in the changelog somewhere :)

Reply:
This is a good point.  My overall thinking was that we want to get to a point where contract validation is on by default rather than off (just opened #1296 on that), so this checking would also happen by default.  For now, though, I wonder if we should do the malformedness checking by default, to avoid weird surprises like the one you mention.  @lazaroclapp thoughts?

Reply:
I changed things to run the malformedness checking by default, as you suggested.

When we turn on contract validation by default, yes, that would only affect codebases with `@Contract` in their annotated code.